frontends/election-manager/src/components/smartcard_modal/smartcard_modal.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useContext } from 'react';
+import styled from 'styled-components';
+import { assert } from '@votingworks/utils';
+import { isSuperadminAuth, Modal, Prose } from '@votingworks/ui';
+import { useLocation } from 'react-router-dom';
+
+import { AppContext } from '../../contexts/app_context';
+import { routerPaths } from '../../router_paths';
+
+const ModalContents = styled.div`
+  padding: 1rem;
+`;
+
+type SmartcardModalView =
+  | 'CardDetails'
+  | 'ProgramElectionCard'
+  | 'ProgramSuperAdminCard';
+
+export function SmartcardModal(): JSX.Element | null {
+  const { auth } = useContext(AppContext);
+  assert(isSuperadminAuth(auth));
+
+  const location = useLocation();
+  const onSuperAdminSmartcardsScreen =
```

Comment:
This could be after the `if (!auth.card) {` short-circuit, right?

Reply:
Yup! The definition of `onSuperAdminSmartcardsScreen` can come after the short circuit - will move. (The use of `useLocation()`, however, can't come after the short circuit since React needs to call the same number of hooks on every render)